habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/util/ContainerizeDepsDockerfileHelper.java

```diff
@@ -69,11 +69,14 @@ public static String createContainerStageContentFrom(String template, String anc
      * @param dockerFile the Dockerfile to be updated
      * @param anchorDirectory the anchor directory
      * @param moduleBaseDir the module base directory
+     * @param owner the uid/name to give ownership of the virtual env to within the container
+     * @param builderBaseImage the image to use for bundling the virtual environment
+     * @param finalBaseImage the image to use for running the virtual environment
      * @return updated Dockerfile content
      */
-    public static String updateDockerfileWithContainerStageLogic(File dockerFile, String anchorDirectory, String moduleBaseDir, String owner, String baseImage) {
-        String builderStageContent = ContainerizeDepsDockerfileHelper.createContainerStageContentFrom(BUILDER_STAGE_TEMPLATE, anchorDirectory, moduleBaseDir, owner, baseImage);
-        String finalStageContent = ContainerizeDepsDockerfileHelper.createContainerStageContentFrom(FINAL_STAGE_TEMPLATE, null, null, owner, baseImage);
+    public static String updateDockerfileWithContainerStageLogic(File dockerFile, String anchorDirectory, String moduleBaseDir, String owner, String builderBaseImage, String finalBaseImage) {
+        String builderStageContent = ContainerizeDepsDockerfileHelper.createContainerStageContentFrom(BUILDER_STAGE_TEMPLATE, anchorDirectory, moduleBaseDir, owner, builderBaseImage);
+        String finalStageContent = ContainerizeDepsDockerfileHelper.createContainerStageContentFrom(FINAL_STAGE_TEMPLATE, null, null, owner, finalBaseImage);
```

Comment:
Q: are we override their final image? if there is a startup script in their final image, will that be an issue?

Reply:
Yes we set the final image `FROM`, so they will have to use the one that is in the snippet. (This was a decision made in the last PR.)  Because of the nuance of creating a compatible venv, IMO it's best to force them to think about it.  If they don't like the default final image, they can still customize it but then should also customize the builder base or the venv likely won't work.

As for script or customization, they can easily add more logic _after_ the FINAL block injected by Habushu, including copying over any startup scripts.

Reply:
Discussed offline the various concerns and came to consensus that this is probably a good base for the functionality to start from.  As we get more feedback and see more edge cases, we can fine tune based on that info.